### PR TITLE
[stable/datadog] Add dependency chart overriding values

### PR DESCRIPTION
**Pull request by cw-sakamoto**
Sorry, I forgot to add parameter for kube-state-metrics in https://github.com/kubernetes/charts/pull/2769.

In kube-state-metrics charts ```rbac.create``` default value is ```false```, but in this chart ```rbac.create``` default value is ```true```, so I overwrite the value of ```rbac.create``` in kube-state-metrics chart with ```true``` by ```kube-state-metrics.rbac.create```.

### CODE REVIEW

**Comment by cw-sakamoto** [comment]
@mgoodness @viglesiasce @dhilipkumars Hi, thanks for maintaining charts 👍
I missed kube-state-metrics's rbac parameter when https://github.com/kubernetes/charts/pull/2769, but @mumoshu  pointed out.

**Comment by irabinovitch** [comment]
Thanks @cw-sakamoto, @hkaj and @xvello   who maintain this chart will review and get back to you with feedback.

File: stable/datadog/README.md
Comment (hkaj) [inline stable/datadog/README.md]: `metricsrbac` should be `metrics`

**Review by hkaj — CHANGES_REQUESTED** [review]

File: stable/datadog/README.md
Comment (cw-sakamoto) [reply stable/datadog/README.md]: @hkaj Thank for your comment!
Sorry...I modiefied README in https://github.com/kubernetes/charts/pull/2969/commits/0c6823ff893c5a37476d0343ec4ef0539ca49afd

**Review by hkaj — APPROVED** [review]

**Comment by irabinovitch** [comment]
@mattfarina Looks good here.  We're ready to merge when you have a moment.

File: stable/datadog/values.yaml
Comment (mattfarina) [inline stable/datadog/values.yaml:172]: Does this actually work? The sections in the code look like:

```
{{- end }}
      serviceAccountName: {{ if .Values.rbac.create }}{{ template "datadog.fullname" . }}{{ else }}"{{ .Values.rbac.serviceAccountName }}"{{ end }}
{{ end }}
```

Shouldn't it use something like `.Values.kube-state-metrics.rbac.serviceAccountName` instead?

Comment (cw-sakamoto) [reply stable/datadog/values.yaml:172]: @mattfarina Thanks for your comments.

>.Values.kube-state-metrics.rbac.serviceAccountName instead?

No. .
If I hadn't misunderstand,　```kube-state-metrics.rbac.serviceAccount```  in values.yaml override parameter for kube-state-metrics charts (https://github.com/kubernetes/charts/blob/master/stable/kube-state-metrics/templates/deployment.yaml#L22).


FLY: https://github.com/kubernetes/helm/blob/master/docs/chart_template_guide/subcharts_and_globals.md#overriding-values-from-a-parent-chart

Comment (mattfarina) [reply stable/datadog/values.yaml:172]: Ah, yup. It was that I missed the rbac section in the values.yaml file on my first pass.

**Comment by mattfarina** [comment]
@hkaj When a PR isn't running the `pull-charts-e2e` automatically it's because the PR author isn't a member of the Kubernetes org. Use the `/ok-to-test` command to enable testing on the PR if you think it's ok. If you want the test merged (as in it's looks good enough) you can use `/lgtm`. This will approve it and pass review. When tests all pass and it's LGTM by an approver it will be merged.

**Comment by hkaj** [comment]
@mattfarina ok, thanks for the tip.

File: stable/datadog/values.yaml
Comment (cw-sakamoto) [reply stable/datadog/values.yaml:172]: Thanks.

**Pull request merged**
